src/Api/Validator.php

```diff
@@ -271,6 +278,28 @@ private function checkAssociativeArray($value)
         return true;
     }
 
+    private function checkDocumentType($value)
+    {
+        if (is_array($value)) {
+            $isValid = true;
+            foreach ($value as $key => $val) {
```

Comment:
Added a validation for this, and hope I fixed the whitespace issues.  The validation takes the type of the first key and compares is sequentially; I figured iterating only once is probably better since this is of unknown size.  
The other considered implementation was getting a list of the keys using array_keys(), then filtering that list by the type of the first element, and seeing if the result was empty.

Reply:
This isn't checking the type of the keys is valid for a document, just that they're the same. It should validate [that it's associative](https://github.com/aws/aws-sdk-php/blob/15f6b434b92826a321191bf5b72820934d970580/build/docs/classes/CodeSnippetGenerator.php#L161) or that all keys are strings.